### ISSUE TITLE
LocallyConnected layers default initializer glorot_uniform determines fanning incorrectly

### ISSUE THREAD

**Issue by cnettel** [tensorflow/tensorflow#53688]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): CentOS 7 running NGC tensorflow_21.07-tf2-py3 container
- TensorFlow installed from (source or binary): binary, Nvidia container
- TensorFlow version (use command below): 2.5.0+nv
- Python version: 3.8.10
- CUDA/cuDNN version: 11.5
- GPU model and memory: Nvidia V100, 32 GB

**Describe the current behavior**
The Keras LocallyConnectednD layers all use default kernel_initializer glorot_uniform, just like their Conv counterparts. However, with the default implementation (1), the parameter tensor for e.g. LocallyConnected1D ends up being three-dimensional input-size * kernel-size * filters. In short, this means that the initialization values for a layer are vastly different if you just switch from Conv1D to LocallyConnected1D, to the point where gradients can easily vanish.

**Describe the expected behavior**
While a locally connected layer and a convolutional layer will have different training dynamics, the underlying argument for the Xavier initialization should result in initial tensors of similar magnitude, since the number of terms in the sum for each activation is identical.

GlorotUniform is a wrapper for VarianceScaling. VarianceScaling will use _compute_fans internally, which just stacks additional input dimensions on top in a multiplicative manner (https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/initializers/initializers_v2.py#L1011). This has been confirmed in the master branch, as indicated in the link.

Due to compat reasons, I guess changing the default at this point could be hard, but with a clean slate the expected behavior would be that the default init for Conv layers and LocallyConnected layers would be identical. At this point, maybe a warning and comments in the docs for both LocallyConnected layers and the initializers on how the fanning is actually determined.

**[Contributing](https://www.tensorflow.org/community/contribute)**

- Do you want to contribute a PR? (yes/no): yes
- Briefly describe your candidate solution(if contributing): adding a warning mechanism in the locally connected layers if used with a default initializer. This is not ideal, as even customized code could assume another definition of fanning.

**Standalone code to reproduce the issue**
https://colab.research.google.com/drive/1bLUD52-syj01zjHyVXhGzdYZggLFM-kf

**Comment by mohantym** [comment]
Hi @cnettel ! 
Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues)
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999) . Thanks!
